tests/TaskLineRenderer.test.ts

```diff
@@ -596,3 +598,60 @@ describe('task line rendering', () => {
         );
     });
 });
+
+describe('Visualise HTML', () => {
+    async function renderAndVerifyHTML(task: Task, layoutOptions: LayoutOptions) {
+        const parentRender = await createMockParentAndRender(task, layoutOptions);
+        const taskAsHTML = parentRender.innerHTML
+            .replace(/ data-/g, '\n    data-')
+            .replace(/<span/g, '\n        <span');
+
+        verifyWithFileExtension(taskAsHTML, 'html');
+    }
+
+    const fullTask = TaskBuilder.createFullyPopulatedTask();
+    const minimalTask = fromLine({ line: '- [-] empty' });
+
+    function layoutOptionsFullMode(): LayoutOptions {
+        const layoutOptions = new LayoutOptions();
+
+        // Show every Task field (not entirely true)
+        Object.keys(layoutOptions).forEach((key) => {
+            const key2 = key as keyof LayoutOptions;
+            layoutOptions[key2] = false;
+        });
+
+        return layoutOptions;
+    }
+
+    function layoutOptionsShortMode(): LayoutOptions {
+        const layoutOptions = new LayoutOptions();
+
+        // Show every Task field (not entirely true)
+        Object.keys(layoutOptions).forEach((key) => {
+            const key2 = key as keyof LayoutOptions;
+            layoutOptions[key2] = false;
+        });
```

Comment:
I really like the functions that you've extracted.

In what remains, this repetition is quite noticeable ... I think it would be worth trying to make it call the FullMode one - and then set the short-mode flag afterwards...

Reply:
done